src/main/java/pix/data/MyData.java

```diff
@@ -0,0 +1,136 @@
+package pix.data;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import pix.tasks.Deadline;
+import pix.tasks.Event;
+import pix.tasks.Task;
+import pix.tasks.ToDo;
+
+/**
+ * MyDate class to keep track of the list of tasks.
+ */
+public class MyData {
+    /** Arraylist to keep track of all tasks. */
+    private final ArrayList<Task> data = new ArrayList<Task>();
+
+    /**
+     * Gets tasks at index.
+     *
+     * @param index Index of task to retrieve.
+     * @return the Task object.
+     */
+    public Task getData(int index) {
+        return this.data.get(index);
+    }
+
+    /**
+     * Delete the task at index.
+     *
+     * @param index Index of task to delete.
+     */
+    public void deleteData(int index) {
+        this.data.remove(index);
+    }
+
+    /**
+     * Adds task to the list of tasks.
+     *
+     * @param command Task to add to the list.
+     */
+    public void setData(Task command) {
+        data.add(command);
+    }
+
+    /**
+     * Marks the task at index as done.
+     *
+     * @param index Index to mark task as done.
+     */
+    public void markDone(int index) {
+        this.data.get(index).markDone();
+    }
+
+
+    /**
+     * Marks the task at index as undone.
+     *
+     * @param index Index to mark task as undone.
+     */
+    public void markUndone(int index) {
+        this.data.get(index).markUndone();
+    }
+
+    /**
+     * Gets the length of the list of tasks.
+     *
+     * @return Length of list.
+     */
+    public int len() {
+        return this.data.size();
+    }
+
+    /**
+     * Saves the list of tasks to a text file on local device.
+     */
+    public void saveToFile() {
+        String home = System.getProperty("user.dir");

```

Comment:
Variable name can be more specific instead of something generic like home, possibly name it like homeDirectory?